tmt/steps/__init__.py

```diff
@@ -1876,11 +1878,30 @@ def prune(self, logger: tmt.log.Logger) -> None:
 
         if self.workdir is None:
             return
-        logger.debug(f"Remove '{self.name}' workdir '{self.workdir}'.", level=3)
-        try:
-            shutil.rmtree(self.workdir)
-        except OSError as error:
-            logger.warning(f"Unable to remove '{self.workdir}': {error}")
+        preservable_members = [
+            member
+            for member in self.workdir.iterdir()
+            if member.name in self._preserved_workdir_members
+        ]
+        removable_members = [
+            member for member in self.workdir.iterdir() if member not in preservable_members
+        ]
+        if preservable_members:
+            for member in removable_members:
+                logger.debug(f"Remove '{member}'.", level=3)
```

Comment:
Again, apologies. It is wrong from me to comment on perfectly fine code. It's just that it links to the above set/link and that I think we can do one less iterdir() and avoid intermediary list, if that make sense. 

Also wanted to to ask about `unlink(missing_ok=True)` parameter, which was added in 3.8, but then I realized FileNotFoundError is being caught by the OSError, so...nevermind.

```suggestion

        workdir_members = self.workdir.iterdir()

        preservable_members = (
            member for member in workdir_members if member.name in self._preserved_workdir_members
        )

        if preservable_members:
            for member in workdir_members:
                if member in preservable_members:
                    continue

```

Reply:
Ha! I've overoptimized it. Thanks!
> we can do list(workdir.iterdir())

or, well, `set(workdir.iterdir())`, right? When order of elements does not matter. /half-joking :) 
fwiw, I think the memory overhead wouldn't be worth in majority of cases, not going to fire up profiler though.

@happz Big thanks for the write-up. 100% with you on all points. I wouldn't bother you all if these file operations weren't so common in tmt and if I understood it well enough. (same with ssh ;) )